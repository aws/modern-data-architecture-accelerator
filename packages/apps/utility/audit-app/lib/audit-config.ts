/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from '@aws-caef/app';
import { BucketInventoryProps } from '@aws-caef/audit-l3-construct';
import { CaefRoleRef } from '@aws-caef/iam-role-helper';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';

import * as configSchema from './config-schema.json';

export interface AuditConfigContents extends CaefBaseConfigContents {
    /**
     * List of refs to roles which will be provided read access to the audit logs
     */
    readRoles?: CaefRoleRef[]
    /**
     * List of source accounts from which audit logs will be accepted
     */
    sourceAccounts?: string[]
    /**
     * List of source regions from which audit logs will be accepted
     */
    sourceRegions?: string[]
    /**
     * Specifies the S3 prefix where inventory data will be accepted on the audit bucket
     */
    inventoryPrefix?: string
    /**
     * The list of expected inventories
     */
    inventories?: BucketInventoryProps[]
}

export class AuditConfigParser extends CaefAppConfigParser<AuditConfigContents> {

    public readonly readRoleRefs: CaefRoleRef[]
    public readonly sourceAccounts: string[]
    public readonly sourceRegions: string[]
    public readonly inventoryPrefix: string
    public readonly inventories?: BucketInventoryProps[]

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )

        this.readRoleRefs = this.configContents.readRoles ? this.configContents.readRoles : []
        this.sourceAccounts = this.configContents.sourceAccounts ? this.configContents.sourceAccounts : []
        this.sourceRegions = this.configContents.sourceRegions ? this.configContents.sourceRegions : []
        this.inventoryPrefix = this.configContents.inventoryPrefix ? this.configContents.inventoryPrefix : "inventory/"
        this.inventories = this.configContents.inventories
    }

}

