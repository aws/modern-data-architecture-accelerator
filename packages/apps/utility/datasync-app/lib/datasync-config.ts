/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from '@aws-caef/app';
import { AgentProps, AgentWithNameProps, LocationNfsProps, LocationObjectStorageProps, LocationS3Props, LocationsByTypeWithNameProps, LocationSmbProps, TaskProps, TaskWithNameProps, VpcProps } from '@aws-caef/datasync-l3-construct';

import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';


import * as configSchema from './config-schema.json';

export interface LocationsByTypeConfig {
    readonly s3?: { [ name: string ]: LocationS3Props }
    readonly smb?: { [ name: string ]: LocationSmbProps }
    readonly nfs?: { [ name: string ]: LocationNfsProps }
    readonly objectStorage?: { [ name: string ]: LocationObjectStorageProps }
}

export interface DataSyncConfigContents extends CaefBaseConfigContents {
    readonly vpc?: VpcProps
    readonly agents?: { [ name: string ]: AgentProps }
    readonly locations?: LocationsByTypeConfig
    readonly tasks?: { [ name: string ]: TaskProps }
}

export class DataSyncConfigParser extends CaefAppConfigParser<DataSyncConfigContents> {
    public readonly vpc?: VpcProps
    public readonly agents?: AgentWithNameProps[]
    public readonly locations?: LocationsByTypeWithNameProps
    public readonly tasks?: TaskWithNameProps[]

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.vpc = this.configContents.vpc
        this.agents = Object.entries( this.configContents.agents || {} ).map( x => {
            return {
                agentName: x[ 0 ],
                ...x[ 1 ]
            }
        } )

        this.locations = {
            s3: Object.entries( this.configContents.locations?.s3 || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } ),
            smb: Object.entries( this.configContents.locations?.smb || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } ),
            nfs: Object.entries( this.configContents.locations?.nfs || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } ),
            objectStorage: Object.entries( this.configContents.locations?.objectStorage || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } )
        }

        this.tasks = Object.entries( this.configContents.tasks || {} ).map( x => {
            return {
                name: x[ 0 ],
                ...x[ 1 ]
            }
        } )
    }
}

