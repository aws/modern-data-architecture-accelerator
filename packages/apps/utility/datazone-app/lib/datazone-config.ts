/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from '@aws-caef/app';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';


import * as configSchema from './config-schema.json';
import { NamedDomainsProps } from '@aws-caef/datazone-l3-construct';

export interface DataZoneConfigContents extends CaefBaseConfigContents {
    readonly domains: NamedDomainsProps
}

export class DataZoneConfigParser extends CaefAppConfigParser<DataZoneConfigContents> {
    public domains: NamedDomainsProps
    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.domains = this.configContents.domains
    }

}

