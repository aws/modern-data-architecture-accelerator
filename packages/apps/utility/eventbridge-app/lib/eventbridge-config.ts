/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParser, MdaaAppConfigParserProps, MdaaBaseConfigContents } from '@aws-mdaa/app';
import { NamedEventBusProps } from '@aws-mdaa/eventbridge-l3-construct';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';


import * as configSchema from './config-schema.json';

export interface EventBridgeConfigContents extends MdaaBaseConfigContents {
    readonly eventBuses: NamedEventBusProps
}

export class EventBridgeConfigParser extends MdaaAppConfigParser<EventBridgeConfigContents> {
    public eventBuses: NamedEventBusProps
    constructor( stack: Stack, props: MdaaAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.eventBuses = this.configContents.eventBuses
    }

}

