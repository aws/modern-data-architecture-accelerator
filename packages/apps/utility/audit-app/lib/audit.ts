/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { AuditL3Construct, AuditL3ConstructProps } from '@aws-caef/audit-l3-construct';
import { CaefAppConfigParserProps, CaefCdkApp } from '@aws-caef/app';
import { CaefL3ConstructProps } from '@aws-caef/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { AuditConfigParser } from './audit-config';


export class AuditCDKApp extends CaefCdkApp {
    constructor( props: AppProps = {} ) {
        super( "audit", props )
    }
    protected subGenerateResources ( stack: Stack, l3ConstructProps: CaefL3ConstructProps, parserProps: CaefAppConfigParserProps ) {

        const appConfig = new AuditConfigParser( stack, parserProps )
        const constructProps: AuditL3ConstructProps = {
            ...{
                sourceAccounts: appConfig.sourceAccounts,
                sourceRegions: appConfig.sourceRegions,
                readRoleRefs: appConfig.readRoleRefs,
                bucketInventories: appConfig.inventories,
                inventoryPrefix: appConfig.inventoryPrefix
            }, ...l3ConstructProps
        }

        new AuditL3Construct( stack, "audit", constructProps );
        return [ stack ]

    }
}

