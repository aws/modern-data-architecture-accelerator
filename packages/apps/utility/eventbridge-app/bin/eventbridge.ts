#!/usr/bin/env node
/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
import { EventBridgeCDKApp } from '../lib/eventbridge';
new EventBridgeCDKApp().generateStack();
