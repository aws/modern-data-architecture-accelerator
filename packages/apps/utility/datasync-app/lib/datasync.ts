/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { DataSyncL3Construct, DataSyncL3ConstructProps } from '@aws-mdaa/datasync-l3-construct';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { DataSyncConfigParser } from './datasync-config';

export class DataSyncCDKApp extends MdaaCdkApp {
    constructor( props: AppProps = {} ) {
        super( props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`) )
    }

    protected subGenerateResources ( stack: Stack, l3ConstructProps: MdaaL3ConstructProps, parserProps: MdaaAppConfigParserProps ) {
        const appConfig = new DataSyncConfigParser( stack, parserProps )
        const constructProps: DataSyncL3ConstructProps = {
            vpc: appConfig.vpc,
            agents: appConfig.agents,
            locations: appConfig.locations,
            tasks: appConfig.tasks,
            ...l3ConstructProps
        }
        new DataSyncL3Construct( stack, "datasync", constructProps );
        return [ stack ]
    }
}