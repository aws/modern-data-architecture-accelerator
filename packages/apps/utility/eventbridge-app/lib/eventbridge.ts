/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { EventBridgeL3Construct, EventBridgeL3ConstructProps } from '@aws-mdaa/eventbridge-l3-construct';
import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { EventBridgeConfigParser } from './eventbridge-config';

export class EventBridgeCDKApp extends MdaaCdkApp {
  constructor(props: AppProps = {}) {
    super(props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`));
  }
  protected subGenerateResources(
    stack: Stack,
    l3ConstructProps: MdaaL3ConstructProps,
    parserProps: MdaaAppConfigParserProps,
  ) {
    const appConfig = new EventBridgeConfigParser(stack, parserProps);
    const constructProps: EventBridgeL3ConstructProps = {
      eventBuses: appConfig.eventBuses,
      ...l3ConstructProps,
    };

    new EventBridgeL3Construct(stack, 'eventbridge', constructProps);
    return [stack];
  }
}
