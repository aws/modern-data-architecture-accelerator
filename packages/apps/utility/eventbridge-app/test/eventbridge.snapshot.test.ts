/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { describe } from '@jest/globals';
import { snapShotTest, snapShotTestApp, Create } from '@aws-mdaa/testing';
import { EventBridgeCDKApp } from '../lib/eventbridge';
import * as path from 'path';

describe('eventbridge Snapshot Tests', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer({
      test: (val: unknown) => typeof val === 'string' && val.includes('[CONFIG:') && val.includes('test-config.yaml]'),
      print: (val: unknown) => {
        const stringVal = val as string;
        return `"${stringVal.replace(/\[CONFIG:[^[\]]*test-config\.yaml\]/, '[CONFIG:test-config.yaml]')}"`;
      },
    });
  });
  snapShotTest(
    'Eventbridge Stack',
    Create.stackProvider(
      'EventbridgeStackMain',
      (_, context) => {
        const moduleApp = new EventBridgeCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        return moduleApp.generateStack();
      },
      {
        module_name: 'test-eventbridge-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );

  snapShotTestApp(
    'Eventbridge App',
    Create.appProvider(
      context => {
        const moduleApp = new EventBridgeCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        moduleApp.generateStack();
        return moduleApp;
      },
      {
        module_name: 'test-eventbridge-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );
});
