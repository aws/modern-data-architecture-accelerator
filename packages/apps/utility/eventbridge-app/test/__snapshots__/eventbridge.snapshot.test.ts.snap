// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eventbridge Snapshot Tests Eventbridge App App Snapshot Test 1`] = `
{
  "test-org-test-env-test-domain-test-eventbridge-main": {
    "Description": "(SO0320-eventbridge) Modern Data Architecture Accelerator. Version 1.0.0",
    "Outputs": {
      "REPLACED-CFN-LOGICAL-ID": {
        "Export": {
          "Name": "test-org:test-domain:test-eventbridge-main:eventbus:testeventbus:arn",
        },
        "Value": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "Arn",
          ],
        },
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Export": {
          "Name": "test-org:test-domain:test-eventbridge-main:eventbus:testeventbus:name",
        },
        "Value": {
          "Ref": "REPLACED-CFN-LOGICAL-ID",
        },
      },
    },
    "Parameters": {
      "BootstrapVersion": {
        "Default": "/cdk-bootstrap/hnb659fds/version",
        "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
    },
    "Resources": {
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "Stack description parameter to update on version changes",
          "Name": "/test-org/test-domain/test-eventbridge-main/aws-solution",
          "Tags": {
            "mdaa_cdk_app": "eventbridge",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-eventbridge-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "(SO0320-eventbridge) Modern Data Architecture Accelerator. Version 1.0.0",
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Name": "test-org-test-env-test-domain-test-eve--6dfaaf3a",
          "Tags": [
            {
              "Key": "mdaa_cdk_app",
              "Value": "eventbridge",
            },
            {
              "Key": "mdaa_domain",
              "Value": "test-domain",
            },
            {
              "Key": "mdaa_env",
              "Value": "test-env",
            },
            {
              "Key": "mdaa_module_name",
              "Value": "test-eventbridge-main",
            },
            {
              "Key": "mdaa_org",
              "Value": "test-org",
            },
          ],
        },
        "Type": "AWS::Events::EventBus",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "ArchiveName": "test-org-test-env-test-domain-test-eve--1d5f1145",
          "Description": {
            "Fn::Join": [
              "",
              [
                "Archive for ",
                {
                  "Ref": "REPLACED-CFN-LOGICAL-ID",
                },
              ],
            ],
          },
          "EventPattern": {
            "account": [
              {
                "Ref": "AWS::AccountId",
              },
            ],
          },
          "RetentionDays": 100,
          "SourceArn": {
            "Fn::GetAtt": [
              "REPLACED-CFN-LOGICAL-ID",
              "Arn",
            ],
          },
        },
        "Type": "AWS::Events::Archive",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "EventBusName": {
            "Ref": "REPLACED-CFN-LOGICAL-ID",
          },
          "Statement": {
            "Action": "events:PutEvents",
            "Effect": "Allow",
            "Principal": {
              "AWS": "some-arn",
              "Service": "some-service.amazonaws.com",
            },
            "Resource": {
              "Fn::GetAtt": [
                "REPLACED-CFN-LOGICAL-ID",
                "Arn",
              ],
            },
            "Sid": "cdk-test-event-bus-put-events",
          },
          "StatementId": "cdk-test-event-bus-put-events",
        },
        "Type": "AWS::Events::EventBusPolicy",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Name": "/test-org/test-domain/test-eventbridge-main/eventbus/test-event-bus/arn",
          "Tags": {
            "mdaa_cdk_app": "eventbridge",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-eventbridge-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Fn::GetAtt": [
              "REPLACED-CFN-LOGICAL-ID",
              "Arn",
            ],
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Name": "/test-org/test-domain/test-eventbridge-main/eventbus/test-event-bus/name",
          "Tags": {
            "mdaa_cdk_app": "eventbridge",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-eventbridge-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Ref": "REPLACED-CFN-LOGICAL-ID",
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
    },
    "Rules": {
      "CheckBootstrapVersion": {
        "Assertions": [
          {
            "Assert": {
              "Fn::Not": [
                {
                  "Fn::Contains": [
                    [
                      "1",
                      "2",
                      "3",
                      "4",
                      "5",
                    ],
                    {
                      "Ref": "BootstrapVersion",
                    },
                  ],
                },
              ],
            },
            "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          },
        ],
      },
    },
  },
}
`;

exports[`eventbridge Snapshot Tests Eventbridge Stack Snapshot Test 1`] = `
{
  "Description": "(SO0320-eventbridge) Modern Data Architecture Accelerator. Version 1.0.0",
  "Outputs": {
    "REPLACED-CFN-LOGICAL-ID": {
      "Export": {
        "Name": "test-org:test-domain:test-eventbridge-main:eventbus:testeventbus:arn",
      },
      "Value": {
        "Fn::GetAtt": [
          "REPLACED-CFN-LOGICAL-ID",
          "Arn",
        ],
      },
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Export": {
        "Name": "test-org:test-domain:test-eventbridge-main:eventbus:testeventbus:name",
      },
      "Value": {
        "Ref": "REPLACED-CFN-LOGICAL-ID",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "Stack description parameter to update on version changes",
        "Name": "/test-org/test-domain/test-eventbridge-main/aws-solution",
        "Tags": {
          "mdaa_cdk_app": "eventbridge",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-eventbridge-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": "(SO0320-eventbridge) Modern Data Architecture Accelerator. Version 1.0.0",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Name": "test-org-test-env-test-domain-test-eve--6dfaaf3a",
        "Tags": [
          {
            "Key": "mdaa_cdk_app",
            "Value": "eventbridge",
          },
          {
            "Key": "mdaa_domain",
            "Value": "test-domain",
          },
          {
            "Key": "mdaa_env",
            "Value": "test-env",
          },
          {
            "Key": "mdaa_module_name",
            "Value": "test-eventbridge-main",
          },
          {
            "Key": "mdaa_org",
            "Value": "test-org",
          },
        ],
      },
      "Type": "AWS::Events::EventBus",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "ArchiveName": "test-org-test-env-test-domain-test-eve--1d5f1145",
        "Description": {
          "Fn::Join": [
            "",
            [
              "Archive for ",
              {
                "Ref": "REPLACED-CFN-LOGICAL-ID",
              },
            ],
          ],
        },
        "EventPattern": {
          "account": [
            {
              "Ref": "AWS::AccountId",
            },
          ],
        },
        "RetentionDays": 100,
        "SourceArn": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Events::Archive",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "EventBusName": {
          "Ref": "REPLACED-CFN-LOGICAL-ID",
        },
        "Statement": {
          "Action": "events:PutEvents",
          "Effect": "Allow",
          "Principal": {
            "AWS": "some-arn",
            "Service": "some-service.amazonaws.com",
          },
          "Resource": {
            "Fn::GetAtt": [
              "REPLACED-CFN-LOGICAL-ID",
              "Arn",
            ],
          },
          "Sid": "cdk-test-event-bus-put-events",
        },
        "StatementId": "cdk-test-event-bus-put-events",
      },
      "Type": "AWS::Events::EventBusPolicy",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Name": "/test-org/test-domain/test-eventbridge-main/eventbus/test-event-bus/arn",
        "Tags": {
          "mdaa_cdk_app": "eventbridge",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-eventbridge-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "Arn",
          ],
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Name": "/test-org/test-domain/test-eventbridge-main/eventbus/test-event-bus/name",
        "Tags": {
          "mdaa_cdk_app": "eventbridge",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-eventbridge-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": {
          "Ref": "REPLACED-CFN-LOGICAL-ID",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
